### ISSUE TITLE
Automatically re-resolve old AR tickets as Incomplete

### ISSUE THREAD

**Issue by violine1101** [mojira/arisa-kt#38]
It would be a great feature if Arisa would automatically close AR tickets that haven't received an update since a long amount of time (say, one or two years) with an adequate message such as
> This issue has been resolved as **Incomplete** since we didn't get an answer in a reasonable amount of time (1 year or more). If this issue still happens, please check if there already is a newer ticket about this, and if there is not, please create a new one.

This will prevent users from resurrecting old reports that have been unattended for a long time by commenting on the ticket.

This may be a bit complicated to implement since the bot currently only checks for recently updated tickets. It would probably need to be adjusted to check a filter with only old AR tickets in regular intervals.

**Comment by urielsalis** [comment]
I was thinking of a possibility to have modules with custom queries, set in configuration. This would allow this kind of work

We do have to make sure we are not doing the same queries multiple times

**Comment by NeunEinser** [comment]
Yes, I also thought already it would be useful to allow for custom queries per module.
That could also have the advantage that arisa will do everything missed after a downtime or when adding a new module.

I think #25 should be worked on beforehand or in conjunction with that though.

**Comment by urielsalis** [comment]
!depends on #25

**Comment by NeunEinser** [comment]
!depends on #59

**Comment by Marcono1234** [comment]
Could we please discuss this with all moderators again or look at previous discussions? @rsh1's and my worries were that this way reports which were once valid get thrown in the same pool of reports which were never valid in the beginning because for them necessary information was never provided. So if (for whatever reason) we wanted to look at those old reports again, we might have a hard(er) time finding them. Additionally recent clean up of old reports showed that quite a large amount of them were still valid.

I know that users tend to comment on old reports, but for me it appeared this happened mostly on already resolved (non-AR) ones. Is them commenting on old AR reports really such a big issue?

**Comment by violine1101** [comment]
Yes, this should definitely be discussed with all the moderators before it is implemented.

The issue with users commenting on old AR tickets is that we won't ever really notice it. Those tickets are old, so they usually don't really show up in our queues (except for the activityfeed, but you know how unreliable it is). This is why I put "create a new ticket" in my proposed message above. If the ticket is reposted, we will get some attention to the issue again.

And at the end of the day the question is, what is a dormant AR ticket really worth if it has not been updated in years? Sure it would be good to take another look at them, but there are just so many of those that manually doing that is not really feasible.

Perhaps it would be a good idea though to ignore tickets that are not unconfirmed or have more than just a few affected versions, these tickets are more likely to be still valid.

**Comment by Marcono1234** [comment]
Or should AR reports which have not been updated for a long time (e.g. 1 year or more) simply not be automatically reopened? And maybe the bot could instead add a comment so that the commenter knows that they can contact us e.g. through /r/Mojira or Discord if we missed their comment.

**Comment by NeunEinser** [comment]
Another way to mitigate losing old AR tickets would be to have Arisa add a comment like `MEQS_OLD_AR` when closing.

**Comment by NeunEinser** [comment]
Since #67 is done, this should be pretty easy now when configuring a query that searches specifically for AR tickets of a certain age

**Comment by Pokechu22** [comment]
The title says "as Invalid", but the body mentions resolving as Incomplete.  To my understanding, incomplete is what has been used in the past, so the title should probably be changed.

**Comment by violine1101** [comment]
Yes, I meant Incomplete, not Invalid.

**Comment by urielsalis** [comment]
Per discussion in https://<communication tool url>/archives/G0NM0QG14/p1588157564115100 and https://<communication tool url>/archives/G1J24FGTH/p1590002081328200, replacing with #170
